src/services/resource.js

```diff
@@ -206,4 +353,315 @@ module.exports = class resourceHelper {
 			throw error
 		}
 	}
+
+	static async upForReview(user_id, organization_id, roles, queryParams, searchText = '', page, limit) {
+		try {
+			let result = {
+				data: [],
+				count: 0,
+				in_progress_count: 0,
+			}
+			const sort = {
+				sort_by: queryParams.sort_by || common.CREATED_AT,
+				order: queryParams.sort_order?.toUpperCase() === common.SORT_ASC ? common.SORT_ASC : common.SORT_DESC,
+			}
+			let finalResourceIds = []
+			let resourceIdsToBeRemoved = []
+			let resources_ids_notes = {}
+			let inProgressResources = []
+
+			const { sequential: resourceTypesInSequentialReview, parallel: resourceTypesInParallelReview } =
+				await this.fetchResourceReviewTypes(organization_id)
+
+			const userRoleTitles = utils.returnUnique(roles.map((item) => item.title))
+
+			let all_resource_types = [...resourceTypesInSequentialReview, ...resourceTypesInParallelReview]
+
+			if (common.TYPE in queryParams && queryParams[common.TYPE]) {
+				all_resource_types = [queryParams[common.TYPE]]
+			}
+
+			const resourceWiseLevels = await this.fetchResourceReviewLevel(
+				organization_id,
+				userRoleTitles,
+				all_resource_types
+			)
+
+			let uniqueResourceIds = []
+			let uniqueOrganizationIds = [organization_id] //initialize current org id
+
+			// check review resources and find all resources under the reviewer name.
+			const reviewResources = await reviewResourcesQueries.findAll(
+				{
+					reviewer_id: user_id,
+				},
+				['resource_id', 'organization_id']
+			)
+
+			if (reviewResources.length > 0) {
+				// get the unique organization ids from reviewResources table by the reviewer
+				uniqueOrganizationIds = utils.returnUnique([
+					...uniqueOrganizationIds,
+					...reviewResources.map((item) => item.organization_id),
+				])
+			}
+			let reviewsFilter = {}
+			let reviewsResponse = {}
+			const find_and_count = await reviewsQueries.countDistinct(
```

Comment:
@adithyadinesh0412 use same logic if the user pass the status as inprogress

Reply:
Not very clear about this comment. This is solely for showing the count of changes requested resources. In the UI they have a button with a badge showing this number. @priyanka-TL